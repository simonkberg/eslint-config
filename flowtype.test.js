/* eslint-env jest */

'use strict'

const fs = require('fs')
const path = require('path')
const config = require('./flowtype.js')
const { CLIEngine } = require('eslint')

const cli = new CLIEngine({
  ignore: false,
  useEslintrc: false,
  configFile: path.resolve(__dirname, 'flowtype.js'),
})

const testFile = filepath =>
  new Promise((resolve, reject) => {
    fs.readFile(filepath, 'utf-8', (err, file) => {
      if (err) {
        reject(err)
      } else {
        resolve(cli.executeOnText(file, filepath))
      }
    })
  })

expect.addSnapshotSerializer({
  print: val => JSON.stringify(val.replace(__dirname, '<PROJECT_ROOT>')),
  test: val => typeof val === 'string',
})

describe('@simonkberg/eslint-config/flowtype', () => {
  it('should match the snapshot', () => expect(config).toMatchSnapshot())

  it('should load config in eslint', async () => {
    const result1 = await testFile(
      path.resolve(__dirname, '__fixtures__/flowtype.js')
    )

    expect(result1.errorCount).toBe(0)
    expect(result1.warningCount).toBe(0)
  })
})
