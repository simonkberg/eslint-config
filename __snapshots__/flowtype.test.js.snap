// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@simonkberg/eslint-config/flowtype should match the snapshot 1`] = `
Object {
  "extends": Array [
    "<PROJECT_ROOT>/node_modules/eslint-plugin-flowtype/dist/configs/recommended.json",
    "<PROJECT_ROOT>/node_modules/eslint-config-prettier/flowtype.js",
  ],
  "parser": "<PROJECT_ROOT>/node_modules/babel-eslint/lib/index.js",
  "plugins": Array [
    "flowtype",
  ],
  "settings": Object {
    "flowtype": Object {
      "onlyFilesWithFlowAnnotation": true,
    },
  },
}
`;
